cylc/flow/run_modes/skip.py

```diff
@@ -0,0 +1,166 @@
+            'job_runner_name': RunMode.SIMULATION,
+            'platform_name': RunMode.SIMULATION,
+            'submit_status': 0   # Submission has succeeded
+        }
+    )
+    for output in process_outputs(itask):
+        task_job_mgr.task_events_mgr.process_message(itask, INFO, output)
+
+    return True
+
+
+def process_outputs(itask: 'TaskProxy') -> List[str]:
+    """Process Skip Mode Outputs:
+
+    * By default, all required outputs will be generated plus succeeded
+      if success is optional.
+    * The outputs submitted and started are always produced and do not
+      need to be defined in outputs.
+    * If outputs is specified and does not include either
+      succeeded or failed then succeeded will be produced.
+
+    Return:
+        A list of outputs to emit.
+    """
+    result: List[str] = []
+    conf_outputs = itask.tdef.rtconfig['skip']['outputs']
+
+    # Remove started or submitted from our list of outputs:
+    for out in get_unecessary_outputs(conf_outputs):
+        conf_outputs.remove(out)
+
+    # Always produce `submitted` output:
+    result.append(TASK_OUTPUT_SUBMITTED)
+    # (No need to produce `started` as this is automatically handled by
+    # task event manager for jobless modes)
+
+    # Send the rest of our outputs, unless they are succeed or failed,
+    # which we hold back, to prevent warnings about pre-requisites being
+    # unmet being shown because a "finished" output happens to come first.
+    for message in itask.state.outputs.iter_required_messages():
+        trigger = itask.state.outputs._message_to_trigger[message]
+        # Send message unless it be succeeded/failed.
+        if trigger in [TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED]:
+            continue
+        if not conf_outputs or trigger in conf_outputs:
+            result.append(message)
+
+    # Send succeeded/failed last.
+    if TASK_OUTPUT_FAILED in conf_outputs:
+        result.append(TASK_OUTPUT_FAILED)
+    else:
+        result.append(TASK_OUTPUT_SUCCEEDED)
+
+    return result
+
+
+def check_task_skip_config(tdef: 'TaskDef') -> None:
+    """Ensure that skip mode configurations are sane at validation time:
+
+    Args:
+        tdef: of task
+
+    Logs:
+        * Warn that outputs need not include started and submitted as these
+          are always emitted.
+
+    Raises:
+        * Error if outputs include succeeded and failed.
+    """
+    skip_config = tdef.rtconfig.get('skip', {})
+    if not skip_config:
+        return
+    skip_outputs = skip_config.get('outputs', {})
+    if not skip_outputs:
+        return
+
+    # Error if outputs include succeded and failed:
+    if (
+        TASK_OUTPUT_SUCCEEDED in skip_outputs
+        and TASK_OUTPUT_FAILED in skip_outputs
+    ):
+        raise WorkflowConfigError(
+            f'Skip mode settings for task {tdef.name} has'
+            ' mutually exclusive outputs: succeeded AND failed.')
+    LOG.info(f'Task {tdef.name} will be run in skip mode.')
```

Comment:
Not sure this is needed or desirable? The platform will be logged when the task submits.